### DISCUSSION TITLE
Eskalation des PR-Chaos

### DISCUSSION THREAD

**Discussion by emkZero** [importPI19fromDHGE/dhge-pi19-sem5#20] [General]
@Steve2955 @RvNovae @nukerxy @LuisSchaefer @photovoltex @sthinbetween 

Um mal kurz zu ventilieren: ich bin mit dem neuen Status quo der Pull Requests nicht zufrieden und finde den vorherigen Zustand, also den vom letzten Semester besser.

Aktuell schreiben mehrere Leute für Github mit und es gibt 2 bis 3 Versionen pro VL als Branch - meistens von denen unter uns, die direkt einchecken und nicht wie ich, der das dann im Nachgang aufarbeiten würde.
So weit so gut, hat ja alles seine Richtigkeit.

Was ich jetzt schlecht finde ist, dass es dann entsprechend viele PRs gibt, die alle auf ``main`` mergen möchten, von denen alle ihre eigene, mehr oder wenger einzigartige Sichtweise auf die VL haben. Dann sitzt mindestens einer von uns da und vergleicht alle PRs, welche nun die beste wäre und mergt ggf. noch Sachen von den anderen rein. Bääh xD

Das entartet also jedes Mal darin, quasi einen merge Konflikt manuell zu lösen, mit dem schlechten Gewissen hinterher, dass jemand zu kurz kommt und traurig ist.

Ich würde das also gerne zur Diskussion stellen, wie wir vielleicht zu einer besseren Lösung kommen:

- ich persönlich habe kein Problem damit, wenn Ergänzungen auf eine Branch von "mir" direkt committet werden, ggf. mit vorheriger Absprache, Review kann ja trotzdem stattfinden - **wie seht ihr das?**
- ich finde, offensichtliche Fehler können auch gleich von irgendjemanden hinterhercommittet werden
- wir könnten uns absprechen, wer einen LVS-Block "übernimmt" und die PR startet; der Rest von uns kann gern ergänzen

Um meiner Tirade jetzt noch einen Schluss zu geben: ich hoffe, ich sehe das jetzt nicht zu verbissen und ich will auch nicht mit Kacke werfen. Mir ist das aktuell ein bisschen chaotisch, aber ich weiß ja auch gar nicht, wie eure Position hierzu ist. Da freue ich mich aber auf eure Rückmeldung ;) ⬇️

**Comment by jsuesse** [comment]
Punkt 1 ist mir als fast Unbeteiligter (Schande!) egal
Punkt 2&3 finde ich super
Je Fach definieren, wer die hauptsächliche Mitschrift macht, die Struktur vorgibt. Alle anderen ergänzen. Den Prozess des Ergänzens halt noch absprechen

**Comment by RvNovae** [comment]
Denke auch, dass es gut wäre, wenn es für jede LVS einen hauptsächlichen Schreiber gibt und andere dann ergänzen. 
Die Abstimmung muss dann natürlich gescheit funktionieren.

Man könnte vielleicht sogar solche Dinge, wie [Live-Share](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare-pack) o.Ä. in Betracht ziehen. Mit solchen Lösungen sollte man wahrscheinlich vorsichtig sein.
